src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java

```diff
@@ -258,6 +259,19 @@ public void testConnectJsonConverter_MapBigDecimalExceedsMaxPrecision() throws J
     //assert expected.toString().equals(result.toString());
   }
 
+  @Test
+  public void testConnectSimpleHeaderConverter_MapDateAndOtherTypes() throws JsonProcessingException {
```

Comment:
Thank you for adding the test. Overall looks good. 
Can this be a breaking change? 

What does the current  json node look like with `"{\"f1\": \"1970-03-22\", \"f2\": true}";`?

Reply:
@C0urante upon running the test, looks like it never executes the new code path. And the json node doesnt match - failing the test, can you verify?

Reply:
I'm sorry, are you referring to a run of the test with or without the functional changes here?

Reply:
Run of the test. 
I opened one more PR #288 which will run the automated tests.

Reply:
So a run of the test _with_ the functional changes?